moveit_ros/planning_interface/move_group_interface/src/move_group_interface.cpp

```diff
@@ -1336,26 +1402,49 @@ class MoveGroupInterface::MoveGroupInterfaceImpl
 moveit::planning_interface::MoveGroupInterface::MoveGroupInterface(const std::string& group_name,
                                                                    const boost::shared_ptr<tf::Transformer>& tf,
                                                                    const ros::WallDuration& wait_for_servers)
+  : MoveGroupInterface(Options(group_name), tf, wait_for_servers)
 {
-  if (!ros::ok())
-    throw std::runtime_error("ROS does not seem to be running");
-  impl_ = new MoveGroupInterfaceImpl(Options(group_name), tf ? tf : getSharedTF(), wait_for_servers);
+  // delegates to the Options, tf, WallDuration constructor
 }
 
+// deprecated
 moveit::planning_interface::MoveGroupInterface::MoveGroupInterface(const std::string& group,
                                                                    const boost::shared_ptr<tf::Transformer>& tf,
                                                                    const ros::Duration& wait_for_servers)
   : MoveGroupInterface(group, tf, ros::WallDuration(wait_for_servers.toSec()))
 {
 }
 
+moveit::planning_interface::MoveGroupInterface::MoveGroupInterface(const MoveGroupInterface& other)
+  : remembered_joint_values_(other.remembered_joint_values_), impl_(new MoveGroupInterfaceImpl(*other.impl_))
+{
+}
+
+moveit::planning_interface::MoveGroupInterface& moveit::planning_interface::MoveGroupInterface::
+operator=(const MoveGroupInterface& other)
+{
+  if (&other == this)
+    return *this;
+
+  this->remembered_joint_values_ = other.remembered_joint_values_;
+  if (this->impl_)
+  {
+    delete this->impl_;
+    impl_ = new MoveGroupInterfaceImpl(*other.impl_);
+  }
+  return *this;
+}
+
 moveit::planning_interface::MoveGroupInterface::MoveGroupInterface(const Options& opt,
                                                                    const boost::shared_ptr<tf::Transformer>& tf,
                                                                    const ros::WallDuration& wait_for_servers)
 {
+  if (!ros::ok())
+    throw std::runtime_error("ROS does not seem to be running");
```

Comment:
is this necessary? if it is, it seems every ros node should have this sort of logic. doesn't ROS already report you forgot to launch roscore even without this?

Reply:
This code already existed in another constructor. I changed that constructor to delegate to this one and moved this logic as well.

Reply:
While I do like 1 as a solution (with move semantics, ideally), I don't think its necessarily problematic. If you made two move group interfaces on different groups, you'd get a similar situation. Nodes communicating with the original source instance will continue to work.